### PR TITLE
[FEATURE][locator] Add delayed fetching of results

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR implements a delayed fetching of results for the locator filters. This comes in handy when spamming a locator filter at every keystroke is either overly CPU intensive or breaks usage policy.

### CODE REVIEW

**Comment by 3nids** [comment]
Rather than implementing the delay in the locator itself, why not implementing this in the connection between the key stroke and the locator? 
You could then directly set a 200 ms delay on each key stroke (except return key)?

**Comment by nirvn** [comment]
@3nids , because some locator filters benefits from instance result fetching.

The practical goal here is to implement a nominatim geocoder locator filter that has a relatively long delay (e.g. 1.5sec).

**Review by nyalldawson — APPROVED** [review]
LGTM!

**Comment by nyalldawson** [comment]
Looks good to me, but let's wait for @3nids feedback before merge

**Comment by nirvn** [comment]
@nyalldawson , thanks -- had some more chit-chat with @3nids yesterday. I'll let him confirm he's OK :wink:

**Review by 3nids — APPROVED** [review]
All good!

**Pull request merged**
